torch/_subclasses/fake_tensor.py

```diff
@@ -148,15 +148,30 @@ def unset_fake_temporarily():
             torch._C._set_dispatch_mode(old)
 
 
+def get_plain_tensors(subclass):
+    assert is_traceable_wrapper_subclass(subclass)
+    plain_tensors = []
+    todo = [subclass]
+    while todo:
+        curr = todo.pop()
+        inner_keys, _ = curr.__tensor_flatten__()
+        for key in inner_keys:
+            val = getattr(curr, key)
+            if not is_traceable_wrapper_subclass(val):
+                plain_tensors.append(val)
+            else:
+                todo.append(val)
+    return plain_tensors
+
+
 def is_fake(x):
     if isinstance(x, FakeTensor):
         return True
     if is_traceable_wrapper_subclass(x):
-        attrs, _ = type(x).__tensor_flatten__(x)
-        flattened_tensors = [getattr(x, attr) for attr in attrs]
-        # need to recurse because we could have nested subclasses
-        all_fake = all(is_fake(x) for x in flattened_tensors)
-        any_fake = any(is_fake(x) for x in flattened_tensors)
+        plain_tensors = get_plain_tensors(x)
```

Comment:
just curious - it seems like the existing version here (prior to your change) was recursive, and already supported nested subclasses? Any reason for the change?

Reply:
Yep no idea why i did it this way lol